src/test/java/io/github/download/maven/plugin/internal/WGetMojoTest.java

```diff
@@ -696,6 +697,29 @@ public void testRetriedAfterDownloadFailsWithCode500() {
         }
     }
 
+    /**
+     * Plugin execution should fail only after all retries have been exhausted
+     * if a custom configured code was returned by the resource being downloaded.
+     * It should repeat the query exactly 3 times.
+     */
+    @Test
+    public void testRetriedAfterDownloadFailsWithCode404() {
+        this.wireMock.stubFor(get(anyUrl()).willReturn(notFound()));
+        try {
+            createMojo(m -> {
+                setVariableValueToObject(m, "uri", URI.create(wireMock.baseUrl()));
+                setVariableValueToObject(m, "skipCache", true);
+                setVariableValueToObject(m, "failOnError", true);
+                setVariableValueToObject(m, "retries", 3);
+                setVariableValueToObject(m, "retriesOnHttpCodes", "404,500");
+            }).execute();
+            fail("The mojo should have failed upon error");
+        } catch (Exception e) {
+            assertThat(e, is(instanceOf(MojoExecutionException.class)));
+            verify(3, getRequestedFor(anyUrl()));
+        }
+    }
+
```

Comment:
@guerrillalg it would be really awesome to have a test(s) which uses all new properties.